packages/patternfly-4/react-core/src/components/Nav/Nav.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import styles from '@patternfly/patternfly-next/components/Nav/styles.css';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /** NavList elements */
+  children(props, propName, componentName) {
+    const prop = props[propName];
+
+    let error = null;
+    React.Children.forEach(prop, child => {
+      if (child.type.componentType !== 'NavList') {
+        error = new Error(`${componentName} children should be of type NavList.`);
+      }
+    });
+    return error;
+  },
+  /** Additional classes added to the Nav */
+  className: PropTypes.string,
+  /** Non-visible label to describe the nav */
+  ariaLabel: PropTypes.string
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  ariaLabel: ''
+};
+
+const renderLists = (children, updateActive, activeGroup, activeItem) =>
+  React.Children.map(children, (childNode, index) =>
+    React.cloneElement(childNode, {
+      groupId: index,
+      updateActive,
+      activeGroup,
+      activeItem
+    })
+  );
+
+class Nav extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeItem: -1,
+      activeGroup: -1
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const { activeItem, activeGroup } = this.state;
+    if (activeItem === nextState.activeItem && activeGroup === nextState.activeGroup) {
+      return false;
+    }
+    return true;
+  }
+
+  // Called back from NavListItem
+  updateActive(groupId, itemId) {
+    this.setState({
```

Comment:
I don't think we should hold which group and item is active, but rather send these information to consumer so he can react on these events, set corresponding active item and this component will be re-rendered.

With this approach the navigation will be activated no matter what. But what if consumer wants to ignore such thing (maybe there is form which needs to be filled before changing the navigation, or consumer has to check if user has access to such nav item)